api/reference/configuration_script_payloads.md

```diff
@@ -0,0 +1,394 @@
+    },
+    {
+      "name": "edit",
+      "method": "patch",
+      "href": "http://localhost:3000/api/configuration_script_payloads/1"
+    },
+    {
+      "name": "edit",
+      "method": "put",
+      "href": "http://localhost:3000/api/configuration_script_payloads/1"
+    }
+  ]
+}
+```
+
+### Editing configuration script payloads
+
+Configuration script payloads can be edited by using a POST to the configuration script payloads collection:
+
+    POST /api/configuration_script_payloads
+
+
+POST /api/configuration_script_payloads
+
+``` json
+{
+  "action": "edit",
+  "resources": [
+    {"id": 1, "description": "A Helpful Description"}
+  ]
+}
+```
+
+Example response:
+
+``` json
+{
+  "results": [
+    {
+      "href": "http://localhost:3000/api/configuration_script_payloads/1",
+      "description": "A Helpful Description",
+      "credentials": {
+        "api_user.$": {
+          "credential_ref": "manageiq_api",
+          "credential_field": "userid"
+        },
+        "api_password.$": {
+          "credential_ref": "manageiq_api",
+          "credential_field": "password"
+        }
+      },
+      "id": "1",
+      "manager_id": "2",
+      "manager_ref": null,
+      "name": "provision-vm-service/list-providers.asl",
+      "variables": null,
+      "created_at": "2023-11-08T17:43:30Z",
+      "updated_at": "2023-11-15T20:14:28Z",
+      "survey_spec": null,
+      "inventory_root_group_id": null,
+      "type": "ManageIQ::Providers::Workflows::AutomationManager::Workflow",
+      "parent_id": null,
+      "configuration_script_source_id": "2",
+      "run_by_userid": null,
+      "payload": "{\n  \"Comment\": \"List Providers.\",\n  \"StartAt\": \"ListProviders\",\n  \"States\": {\n    \"ListProviders\": {\n      \"Type\": \"Task\",\n      \"Resource\": \"docker://docker.io/agrare/list-providers:latest\",\n      \"End\": true,\n      \"Credentials\": {\n        \"api_user.$\": \"$.api_user\",\n        \"api_password.$\": \"$.api_password\",\n        \"api_token.$\": \"$.api_token\",\n        \"api_bearer_token.$\": \"$.api_bearer_token\"\n      },\n      \"Parameters\": {\n        \"API_URL.$\": \"$$.Execution._manageiq_api_url\",\n        \"VERIFY_SSL.$\": \"$.dialog.dialog_verify_ssl\",\n\n        \"PROVIDER_TYPE\": \"ManageIQ::Providers::Vmware::InfraManager\"\n      }\n    }\n  }\n}\n",
+      "payload_type": "json",
+      "context": null,
+      "output": null,
+      "status": null,
+      "miq_task_id": null,
+      "payload_valid": true,
+      "payload_error": null
+    }
+  ]
+}
+```
+
+
+For instance, configuration script payloads can be edited to map credentials:
+
+    POST /api/configuration_script_payloads
+
+``` json
+{
+  "action": "edit",
+  "resources": [
+    {
+      "id": 1,
+      "credentials": [
+        {"my-cred-user": {"credential_ref": "my-credential", "credential_field": "userid"},
+        {"my-cred-password": {"credential_ref": "my-credential", "credential_field": "password"}
```

Comment:
Looks like a typo in the suggestion :wink: 

```suggestion
        {"my-cred-user": {"credential_ref": "my-credential", "credential_field": "userid"}},
        {"my-cred-password": {"credential_ref": "my-credential", "credential_field": "password"}}
```

Reply:
I have fixed this typo, Thanks!